### ISSUE TITLE
any upcoming release of pyment?

### ISSUE THREAD

**Issue by sunweaver** [dadadel/pyment#45]
Hi,

my name is Mike Gabriel and I am a Debian Developer, that is, I can upload software to Debian.

I stumbled over this project when in need for a simply epydoc -> sphinx docstring converter.  As I got fond of this nice tool, I consider bringing pyment to Debian.

For bringing pyment to Debian, it would be nice to have a recent release (not one that is 3 years old). Of course, I can package a Git snapshot, but a release tag is much more preferred, so that's why I ask.

Thanks!
Mike

**Comment by dadadel** [comment]
Hi @sunweaver,

Thanks for your interest on Pyment. It would be great to see it in Debian of course!
I plan to release a new version asap I will fix some issues so I'll tell you when you can start the work.

Cheers
Adel

**Comment by sunweaver** [comment]
Hi,

On Wednesday, September 27, 2017, Daouzli wrote:
> Hi @sunweaver,
> Thanks for your interest on Pyment. It would be great to see it in Debian of course!
> I plan to release a new version asap I will fix some issues so I'll tell you when you can start the work.

Great, don't let me wait too long, as I need pyment for migrating one of my projects. ;-) The requirement here is to have the software packaged.

Thanks,
Mike

-- 
Sent from my Fairphone (running Sailfish OS)...

**Comment by dadadel** [comment]
Ok @sunweaver,
Let me know, in worth case, when you'd like to have a release?

**Comment by sunweaver** [comment]
Hi,

On  Do 28 Sep 2017 09:46:25 CEST, Daouzli wrote:

> Ok @sunweaver,
> Let me know, in worth case, when you'd like to have a release?

Basically, as soon as possible. But it needs to fit in your agenda, too.

THANKS!
Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by dadadel** [comment]
Hi Mike,
Is it okay for a release if some tests are KO @sunweaver ?

**Comment by sunweaver** [comment]
Let me know what tests fail and I can also take a look...
Mike

On Monday, October 2, 2017, Daouzli wrote:
> Hi Mike,
> Is it okay for a release if some tests are KO @sunweaver ? 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.￼
>

-- 
Sent from my Fairphone (running Sailfish OS)...

**Comment by dadadel** [comment]
FAIL: testIssue34 (test_issues.IssuesTests)
FAIL: testIssue46 (test_issues.IssuesTests)
FAIL: testIssue47 (test_issues.IssuesTests)
FAIL: testCaseNoGenDocsAlreadyGoogle (test_pyment_cases.FilesConversionTests)
FAIL: testCaseNoGenDocsAlreadyNumpydoc (test_pyment_cases.FilesConversionTests)

**Comment by sunweaver** [comment]
HI,

On  Mo 02 Okt 2017 16:33:35 CEST, Daouzli wrote:

> FAIL: testIssue34 (test_issues.IssuesTests)
> FAIL: testIssue46 (test_issues.IssuesTests)
> FAIL: testIssue47 (test_issues.IssuesTests)
> FAIL: testCaseNoGenDocsAlreadyGoogle (test_pyment_cases.FilesConversionTests)
> FAIL: testCaseNoGenDocsAlreadyNumpydoc  
> (test_pyment_cases.FilesConversionTests)

this might be a red herring, but...

```
diff --git a/pyment/pyment.py b/pyment/pyment.py
index cc479a4..f2a55b3 100644
--- a/pyment/pyment.py
+++ b/pyment/pyment.py
@@ -116,6 +116,7 @@ class PyComment(object):
                                output_style=self.output_style,
                                first_line=self.first_line,
                                **self.kwargs)
+                print (e.get_input_style())
                  elem_list.append({'docs': e, 'location': (-i, -i)})
              else:
                  if waiting_docs and ('"""' in l or "'''" in l):
```

I added the above line to see what input style gets detecting during  
the tests.

It most of the time shows javadoc style, which seems incorrect to me.

As the tests should be run at package build time, they should better  
work. All of them.

Let me know, if I am correct with the broken autodetect_style() method  
or if I need to search deeper.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by dadadel** [comment]
@sunweaver,

It is not critical of course and Pyment can work correctly in most cases.
There are also other issues that are not tested yet. The only problem I see is that as I added those tests in the Pyment test suite so there is not 100% tests OK.
I plan to fix them but I'm not sure to be able to do it quickly.
The 5 failing tests concern the issues #34, #46, #47, #48, #49.
They deal with proceeding already existent docstrings that should not be modified, or with particular default param values. Thus, it's not so serious!

**Comment by dadadel** [comment]
Hi @sunweaver,

I've just released version 3.2.0 of Pyment. Sorry to have been so long but I was very busy.
I hope it is not too late. Let me know.

Thanks

Adel

**Comment by sunweaver** [comment]
Hi!

On  Di 17 Okt 2017 21:50:13 CEST, Daouzli wrote:

> Hi @sunweaver,
>
> I've just released version 3.2.0 of Pyment. Sorry to have been so  
> long but I was very busy.
> I hope it is not too late. Let me know.
>
> Thanks
>
> Adel

I'll put pyment on my packaging to-do list just now and will revisit  
the latest release soon.

Thanks,
Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by dadadel** [comment]
Hi @sunweaver 
I will close this issue as it is old!
Note that I will soon release a new version (v0.4.0) in Pypi.
